src/show/components/TextField/index.js

```diff
@@ -1,23 +1,79 @@
-import React from "react";
-
+import React, { useState } from 'react';
 const TextField = ({
+  name,
+  type = 'text',
   label = "",
   placeholder = "",
-  required = false,
   value = "",
-  password = false,
-}) => (
-  <div>
-    <h6>{label}</h6>
-    <input
-      placeholder={placeholder}
-      type={password ? "password" : "text"}
-      password={true}
-      className="bg-gray-100 text-black w-64 px-2 rounded border-2 border-black"
-    //   value={value}
-    />
-    <h6 className="text-xs text-right">{required && "required"}</h6>
-  </div>
-);
+  onChange,
+  hasEye,
+}) => {
+  const [error, setError] = useState('');
+  const [inputType, setInputType] = useState(type);
+
+  const validate = (value) => {
+    if (type === 'email') {
+      const emailPattern = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+      if (!emailPattern.test(value)) {
```

Comment:
const validate = (type, value) => {
  const validationRules = {
    email: {
      pattern: /^[^\s@]+@[^\s@]+\.[^\s@]+$/,
      errorMessage: 'Invalid email address'
    },
    password: {
      pattern: /^(?=.*[0-9])(?=.*[!@#$%^&*])[A-Za-z\d!@#$%^&*]{8,}$/,
      errorMessage: 'Password must be at least 8 characters long and include at least one number and one special character'
    }
  };

  // Check if the type exists in the validation rules
  if (validationRules[type]) {
    const { pattern, errorMessage } = validationRules[type];
    if (!pattern.test(value)) {
      setError(errorMessage);
      return;
    }
  }

  setError(''); 
};

Reply:
i replaced previous code with this one